example/verifyartifact/main.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// This is a simple example of how to verify an artifact
+// attestations hosted on GitHub using the sigstore-go library.
+// This is a very barebones example drawn from the sigstore-go
+// library's examples and should not be used in production.
+
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/google/go-github/v66/github"
+	"github.com/sigstore/sigstore-go/pkg/bundle"
+	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/verify"
+)
+
+var (
+	owner                   = flag.String("owner", "cli", "GitHub organization or user to scope attestation lookup by")
+	artifactDigest          = flag.String("artifact-digest", "2ce2e480e3c3f7ca0af83418d3ebaeedacee135dbac94bd946d7d84edabcdb64", "digest of the artifact")
+	artifactDigestAlgorithm = flag.String("artifact-digest-algorithm", "sha256", "The algorithm used to compute the digest of the artifact")
+	expectedIssuer          = flag.String("expected-issuer", "https://token.actions.githubusercontent.com", "Issuer of the OIDC token")
+	expectedSAN             = flag.String("expected-san", "https://github.com/cli/cli/.github/workflows/deployment.yml@refs/heads/trunk", "The expected Subject Alternative Name (SAN) of the certificate used to sign the attestation")
+	trustedRootJSONPath     = flag.String("trusted-root-json-path", "verifyartifact/trusted-root-public-good.json", "Path to the trusted root JSON file")
```

Comment:
How is this file generated?

Reply:
The json file is a snapshot of the public good trusted root hosted by sigstore that underlies the fulcio CA that GitHub uses to sign the attestations in public repos. 

https://github.com/sigstore/root-signing
https://github.com/sigstore/fulcio

Reply:
Thank you!  This would also be an extremely valuable comment to add to this file.